x/data/client/testsuite/cli_test.go

```diff
@@ -0,0 +1,444 @@
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestTxSignData() {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+
+	cid := s.storedCid
+
+	var commonFlags = []string{
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+	}
+
+	testCases := []struct {
+		name         string
+		args         []string
+		expectErr    bool
+		expectErrMsg string
+		respType     proto.Message
+		expectedCode uint32
+	}{
+		{
+			"wrong signer address",
+			append(
+				[]string{
+					"wrongSigner",
+					cid.String(),
+				},
+				commonFlags...,
+			),
+			true,
+			"The specified item could not be found in the keyring",
+			nil,
+			0,
+		},
+		{
+			"wrong cid",
+			append(
+				[]string{
+					val.Address.String(),
+					"wrongCid",
+				},
+				commonFlags...,
+			),
+			true,
+			"selected encoding not supported",
+			nil,
+			0,
+		},
+		{
+			"correct data",
+			append(
+				[]string{
+					val.Address.String(),
+					cid.String(),
+				},
+				commonFlags...,
+			),
+			false,
+			"",
+			&sdk.TxResponse{},
+			0,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := dataclient.MsgSignDataCmd()
+
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, tc.args)
+			if tc.expectErr {
+				s.Require().Contains(out.String(), tc.expectErrMsg)
+			} else {
+				s.Require().NoError(err, out.String())
+				s.Require().NoError(clientCtx.JSONMarshaler.UnmarshalJSON(out.Bytes(), tc.respType), out.String())
+
+				txResp := tc.respType.(*sdk.TxResponse)
+				s.Require().Equal(tc.expectedCode, txResp.Code, out.String())
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestTxStoreData() {
+	val := s.network.Validators[0]
+	val2 := s.network.Validators[1]
+	clientCtx := val.ClientCtx
+
+	cid := s.storedCid
+	base64Encoded := base64.StdEncoding.EncodeToString(s.testContent)
+
+	var commonFlags = []string{
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+	}
+
+	testCases := []struct {
+		name         string
+		args         []string
+		expectErr    bool
+		expectErrMsg string
+		respType     proto.Message
+		expectedCode uint32
+	}{
+		{
+			"store bad data",
+			append(
+				[]string{
+					val.Address.String(),
+					cid.String(),
+					"wrongBase64",
+				},
+				commonFlags...,
+			),
+			true,
+			"illegal base64 data",
+			nil,
+			0,
+		},
+		{
+			"wrong signer",
+			append(
+				[]string{
+					val2.Address.String(),
+					cid.String(),
+					base64Encoded,
+				},
+				commonFlags...,
+			),
+			true,
+			"The specified item could not be found in the keyring",
+			nil,
+			0,
+		},
+		{
```

Comment:
would like to see a test case where the author stores dummy data (tx should fail, because cid doesn't correspond to data)